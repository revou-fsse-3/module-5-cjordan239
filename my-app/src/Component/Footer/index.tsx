// components/Footer.tsx

import React from 'react';
import styles from './styles/style-footer.module.css';

const Footer = () => {
  return (
    <footer className={styles['footer']}>
      <div>
        <p>&copy; 2024 Your Website Name</p>
      </div>
      <div className={styles['social']}>
        <a href="https://www.facebook.com" target="_blank" rel="noopener noreferrer">
          <img
            src="https://th.bing.com/th/id/OIP.K61w8tCEKaKN--vUwjeSSwHaHa?w=191&h=191&c=7&r=0&o=5&dpr=1.4&pid=1.7"
            alt="Facebook"
            className={styles['icon']}
          />
        </a>
        <a href="https://www.twitter.com" target="_blank" rel="noopener noreferrer">
          <img
            src="https://th.bing.com/th/id/OIP.erK8SN73oi9FYRoCu1ZN8QHaHa?rs=1&pid=ImgDetMain"
            alt="Twitter"
            className={styles['icon']}
          />
        </a>
        <a href="https://www.instagram.com" target="_blank" rel="noopener noreferrer">
          <img
            src="https://th.bing.com/th/id/OIP.-ZirgQE5pr8e7htQWowJIgHaHa?rs=1&pid=ImgDetMain"
            alt="Instagram"
            className={styles['icon']}
          />
        </a>
      </div>
      <div className={styles['donation']}>
        <a href="#" target="_blank" rel="noopener noreferrer">
          Buy me a coffee <img src="#" alt="" />
        </a>
      </div>
    </footer>
  );
};

export default Footer;
