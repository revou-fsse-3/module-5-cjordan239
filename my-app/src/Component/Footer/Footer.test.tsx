import { render, screen } from '@testing-library/react';
import Footer from '.';

describe('Unit test for Footer', () => {
    test('should render footer with social media icons', () => {
        render(<Footer />);
        expect(screen.getByAltText('Facebook')).toBeDefined();
        expect(screen.getByAltText('Twitter')).toBeDefined();
        expect(screen.getByAltText('Instagram')).toBeDefined();
        expect(screen.getByAltText('Facebook')).toMatchSnapshot();
        expect(screen.getByAltText('Twitter')).toMatchSnapshot();
        expect(screen.getByAltText('Instagram')).toMatchSnapshot();
    });

    test('should render footer with "Buy me a coffee" link', () => {
        render(<Footer />);
        expect(screen.getByText('Buy me a coffee')).toBeDefined();
        expect(screen.getByText('Buy me a coffee')).toMatchSnapshot();
    });
});
