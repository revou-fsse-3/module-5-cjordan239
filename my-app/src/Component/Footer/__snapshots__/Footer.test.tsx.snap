// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit test for Footer should render footer with "Buy me a coffee" link 1`] = `
<a
  href="#"
  rel="noopener noreferrer"
  target="_blank"
>
  Buy me a coffee 
  <img
    alt=""
    src="#"
  />
</a>
`;

exports[`Unit test for Footer should render footer with social media icons 1`] = `
<img
  alt="Facebook"
  class="icon"
  src="https://th.bing.com/th/id/OIP.K61w8tCEKaKN--vUwjeSSwHaHa?w=191&h=191&c=7&r=0&o=5&dpr=1.4&pid=1.7"
/>
`;

exports[`Unit test for Footer should render footer with social media icons 2`] = `
<img
  alt="Twitter"
  class="icon"
  src="https://th.bing.com/th/id/OIP.erK8SN73oi9FYRoCu1ZN8QHaHa?rs=1&pid=ImgDetMain"
/>
`;

exports[`Unit test for Footer should render footer with social media icons 3`] = `
<img
  alt="Instagram"
  class="icon"
  src="https://th.bing.com/th/id/OIP.-ZirgQE5pr8e7htQWowJIgHaHa?rs=1&pid=ImgDetMain"
/>
`;
